### PR TITLE
Hash Verify Script/Cmdlet for customers

### PR DESCRIPTION
**Pull request by yil578**
Task 9635671 [OAW] Provide OAW.zip Hash Verify Script/Cmdlet in AzS Tools repo for customers

### CODE REVIEW

File: HashVerify/Verify-Hash.psm1 [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    {
+        Write-Host "SUCCESS: $fileName file hash matches."
+    }
+    else
+    {
+        Write-Error "ERROR: $fileName file hash does not match! It isn't safe to use it, please download it again. Actual hash: $actualHash."
+    }
+}
+
+# SIG # Begin signature block
```
Comment (bganapa) [inline HashVerify/Verify-Hash.psm1:36]: we should not have the signature checked in. pls sign it part of the OAW build preocess

File: HashVerify/Verify-Hash.psm1 [hunk 2]
```diff
@@ -0,0 +1,227 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+
+<#
+This module contains the function to verify expected hash of the file specified in the provided path.
+#>
+
+
+function Verify-Hash {
```
Comment (bganapa) [inline HashVerify/Verify-Hash.psm1:8]: Test-FileHash

File: HashVerify/Verify-Hash.psm1 [hunk 3]
```diff
@@ -0,0 +1,227 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (bganapa) [inline HashVerify/Verify-Hash.psm1:1]: do we have any other module that could fit this cmdlet in to?

File: HashVerify/Verify-Hash.psm1 (hunk 2 above)
Comment (yil578) [reply HashVerify/Verify-Hash.psm1:8]: will change it as Test is an approved verb for cmdlet. Thank you :)

File: HashVerify/Verify-Hash.psm1 (hunk 3 above)
Comment (yil578) [reply HashVerify/Verify-Hash.psm1:1]: Couldn't find another module similar enough to put this in, and I think this function could be independent enough to has its own module

File: HashVerify/Verify-Hash.psm1 [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+        [string]
+        $FilePath
+    )
+
+    $fileName = [System.IO.Path]::GetFileName($FilePath)
+    $actualHash = (Get-FileHash -Path $FilePath).Hash
+    Write-Host "$fileName expected hash: $ExpectedHash"
+    if ($ExpectedHash -eq $actualHash)
+    {
+        Write-Host "SUCCESS: $fileName file hash matches."
```
Comment (bganapa) [inline HashVerify/Verify-Hash.psm1:28]: return $true

File: HashVerify/Verify-Hash.psm1 [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    }
+    else
+    {
```
Comment (bganapa) [inline HashVerify/Verify-Hash.psm1:31]: return $false

File: HashVerify/Verify-Hash.psm1 (hunk 2 above)
Comment (troettinger) [reply HashVerify/Verify-Hash.psm1:8]: Make sure you also update the readme example as you change the verb to test

**Review by troettinger — CHANGES_REQUESTED** [review]
Update Readme to match new verb

File: HashVerify/Verify-Hash.psm1 (hunk 2 above)
Comment (yil578) [reply HashVerify/Verify-Hash.psm1]: have updated

**Review by troettinger — APPROVED** [review]

**Pull request merged**